### ISSUE TITLE
iFrame Security Headers not getting added in response headers

### ISSUE THREAD

**Issue by mohd-baquir-qureshi** [Kyon147/laravel-shopify#55]
*For bug reporting only! If you're posting a feature request or discussion, please ignore.*

## Expected Behavior

Please describe the behavior you are expecting.

"Content-Security-Policy" header should be added in the response.

## Current Behavior

Please describe the current behavior?

"Content-Security-Policy" header is not being added in response.

## Failure Information

Please help provide information about the failure if this is a bug.

I think the iFrame Protection Middleware is not working.

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Completely setup the app.
2. Open the chrome inspector.
3. Open the app.
4. Now check any route of your app and click on that.
5. Now you will see the response headers, In that check "Content-Security-Policy" header, it is not there but it should be present else the app will not be approved.

### Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

* Package Version: v17.4.2
* Laravel Version: v9.19.0
* PHP Version: v8.0.2
* Template Engine: Blade
* Using a toolset (Docker, Laradock, Vagrant, etc.): No

### Failure Logs

Please include any relevant log snippets or files here.

NA

**Comment by Kyon147** [comment]
I assume you are getting an app rejection? 

I've seen this happen with Blade because of the redirect that happens after OAUTH to the authenticate/token route.

There's no current solution apart from making your app run like an SPA using Livewire or similar.

**Comment by mohd-baquir-qureshi** [comment]
Yes I am getting app rejection email.

But I have to create the app in blade only, and for approval it's needed.

But in the previous osiset/laravel-shopify it was working fine and in this
also the iframeprotection middleware is there but still it's not adding the
headers.

**Comment by Kyon147** [comment]
I've created a new PR which seems to fix the issue. If the tests pass, I will release a new patch later on.

https://github.com/Kyon147/laravel-shopify/pull/56

**Comment by joeelia** [comment]
@mohd-baquir-qureshi, if you want to fix this before the patch, you can add this line of code to your **kernel.php** under the 'web' middleware group:

` \Osiset\ShopifyApp\Http\Middleware\IframeProtection::class`

Example:
File: app/Http/Kernel.php
``` 
'web' => [
            \App\Http\Middleware\EncryptCookies::class,
            \Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse::class,
            \Illuminate\Session\Middleware\StartSession::class,
           ............
            \Osiset\ShopifyApp\Http\Middleware\IframeProtection::class, //<----- Add this at the end
        ], 
```

**Comment by joeelia** [comment]
@mohd-baquir-qureshi the PR is out, and you can update the package or you cant use my solution above. If the issue is resolved then close this bug..

**Comment by Kyon147** [comment]
This should be fixed now in `17.4.3` so closing but re-open if it still happens after upgrading.
